In [1]:
import os
import random

used = "used.txt"
if not os.path.exists(used):
    open(used, "w+").close()

with open(used, "r+", encoding="utf8") as f:
    used_problems = f.readlines()
    used_problems = [u.strip("\n") for u in used_problems]
    print("Used:", used_problems)

    r = [i for i in range(1, 719)]
    r = [i for i in r if str(i) not in used_problems]
    random.shuffle(r)
    now = None
    if len(r) == 0:
        print("All done!")
        exit(0)
    elif len(r) == 1:
        now = r[0]
    else:
        now = random.choice(r)
    print("Now:", now)
    f.write(str(now) + "\n")

Used: ['361', '474', '297', '340', '698', '289', '489', '627', '405', '35', '666', '33', '356', '182', '671']
Now: 149


## Max Points on a Line
**Date**: 2018-1-17

**Task**: [149](https://leetcode.com/problems/max-points-on-a-line/description/)

**Detail**:

　Given n points on a 2D plane, find the maximum number of points that lie on the same straight line.